hooks/useGetPaymentStatus.tsx

```diff
@@ -0,0 +1,52 @@
+import React from "react"
+
+import { useSubscription } from "@apollo/client"
+
+import { ACTIONS, ACTIONTYPE } from "../components/ParsePOSPayment/reducer"
+import { OperationError } from "../lib/graphql/index.types.d"
+import { LN_INVOICE_PAYMENT_STATUS } from "../lib/graphql/subscription"
+
+interface Props {
+  paymentRequest: string
+  callbackFn?: () => void
+  dispatch?: React.Dispatch<ACTIONTYPE>
+}
+
+const useGetPaymentStatus = ({ paymentRequest, callbackFn, dispatch }: Props) => {
+  const { loading, error, data } = useSubscription<{
+    lnInvoicePaymentStatus: {
+      errors: OperationError[]
+      status?: string
+    }
+  }>(LN_INVOICE_PAYMENT_STATUS, {
+    variables: {
+      input: {
+        paymentRequest,
+      },
+    },
+    onSubscriptionData: ({ subscriptionData }) => {
+      if (subscriptionData?.data?.lnInvoicePaymentStatus?.status === "PAID") {
+        if (!callbackFn) return
+        callbackFn && callbackFn()
+      }
+    },
+  })
+  if (data) {
+    const { status } = data.lnInvoicePaymentStatus
+    if (dispatch) {
+      if (status === "PAID") {
+        dispatch({ type: ACTIONS.GET_PAYMENT_STATUS, payload: "PAID" })
```

Comment:
Will this enter an infinite loop where once an action is dispatched the hook will be rerendered and will dispatch the action again?

Reply:
Yeah. It causes a loop and I have removed it from the hook. Not sure if I should push the fix in the next PR or just push another commit to this branch.